scripts/upgrade.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/sh -xe
+
+info()
+{
+    echo '[INFO] ' "$@"
+}
+
+fatal()
+{
+    echo '[ERROR] ' "$@" >&2
+    exit 1
+}
+
+verify_system() {
+    if [ -x /host/sbin/openrc-run ]; then
+        HAS_OPENRC=true
+        return
+    fi
+    if [ -d /host/run/systemd ]; then
+        HAS_SYSTEMD=true
+        return
+    fi
+    fatal 'Can not find systemd or openrc to use as a process supervisor for k3s'
+}
+
+setup_verify_arch() {
+    if [ -z "$ARCH" ]; then
+        ARCH=$(uname -m)
+    fi
+    case $ARCH in
+        amd64)
+            ARCH=amd64
+            SUFFIX=
+            ;;
+        x86_64)
+            ARCH=amd64
+            SUFFIX=
+            ;;
+        arm64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        aarch64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        arm*)
+            ARCH=arm
+            SUFFIX=-${ARCH}hf
+            ;;
+        *)
+            fatal "Unsupported architecture $ARCH"
+    esac
+}
+
+get_k3s_process_info() {
+  K3S_PID=$(pgrep k3s-)
```

Comment:
We should check for the existence of pgrep a la `type pgrep` early in the script so that we can give a better failure message if procps is not installed.

Reply:
sure will add that to the script, thanks

Reply:
done